app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,29 @@
+class PostsController < ApplicationController
+  before_action :require_log_in, only: [:new, :create]
+
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(whitelist_params)
+    @post.user_id = current_user.id
+
+    if @post.save
+        redirect_to posts_path
+    else
+        render 'new'
+    end
+  end
+  
+
+  private
+
+  def whitelist_params
```

Comment:
Rather than `whitelist_params`, the convention is to use the specific model name to be a bit clearer, i.e. `post_params`.